libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -781,8 +792,7 @@ void ContractLevelChecker::checkOverrideList(FunctionMultiSet const& _funcSet, F
 	for (auto [begin, end] = _funcSet.equal_range(&_function); begin != end; begin++)
 	{
 		// Validate the override
-		if (!checkFunctionOverride(_function, **begin))
-			break;
```

Comment:
Does this generate duplicate errors now?

Reply:
well, it can print similar errors for overrides, but the secondary error location would then show to a different overloaded function.

Example:

```
abstract contract C
{
	function foo() internal;
}
abstract contract A
{
	function foo() internal;
}
contract D is C, A {
	function foo() internal override(A, C) {}
}

Error: Trying to override non-virtual function. Did you forget to add "virtual"?
 --> test.sol:10:2:
  |
10 |    function foo() internal override(A, C) {}
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Note: Overridden function is here:
 --> test.sol:3:2:
  |
3 |     function foo() internal;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^

Error: Trying to override non-virtual function. Did you forget to add "virtual"?
 --> test.sol:10:2:
  |
10 |    function foo() internal override(A, C) {}
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Note: Overridden function is here:
 --> test.sol:7:2:
  |
7 |     function foo() internal;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^


```

Reply:
Maybe it would be better to swap primary and secondary locations?

Reply:
Ok, maybe the order is correct, but we could move the `Did you forget to add "virtual"` to the secondary location? As it is now, it could suggest that you should add `virtual` to the overriding function.

Reply:
That has already been done here: https://github.com/ethereum/solidity/pull/7647/files#diff-b1ba2bbc8a967e12c1721caf45edb970R283

The example is from an earlier time

Reply:
For clarification, this is how the output with the current code looks:

```
rror: Trying to override non-virtual function. Did you forget to add "virtual"?
 --> test.sol:4:2:
  |
4 |     function foo() internal;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^
Note: Overriding function is here:
  --> test.sol:11:2:
   |
11 |    function foo() internal override(A, C) {}
   |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Trying to override non-virtual function. Did you forget to add "virtual"?
 --> test.sol:8:2:
  |
8 |     function foo() internal;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^
Note: Overriding function is here:
  --> test.sol:11:2:
   |
11 |    function foo() internal override(A, C) {}
   |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^


```